hazelcast-cloud/src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```
@@ -59,6 +66,29 @@ public DescribeInstances(AwsConfig awsConfig, String endpoint) {
         attributes.put("X-Amz-Expires", "30");
     }
 
+    private void getKeysFromIamRole() {
+        try {
+            String query = "latest/meta-data/iam/security-credentials/" + awsConfig.getIamRole();
+            URL url = new URL("http", "169.254.169.254", -1, query);
+            InputStreamReader is = new InputStreamReader(url.openStream(), "UTF-8");
+            BufferedReader reader = new BufferedReader(is);
+            Pattern pattern = Pattern.compile(": \"(.*?)\",");
+            for (String line; (line = reader.readLine()) != null; ) {
+                Matcher matcher = pattern.matcher(line);
+                if (line.contains("AccessKeyId") && matcher.find()) {
+                    awsConfig.setAccessKey(matcher.group(1));
+                }
+                if (line.contains("SecretAccessKey") && matcher.find()) {
+                    awsConfig.setSecretKey(matcher.group(1));
+                }
+                if (line.contains("Token") && matcher.find()) {
+                    attributes.put("X-Amz-Security-Token", matcher.group(1));
+                }
+            }
+        } catch (IOException io) {

```

empty catch.. we should fail-fast here.